### PR TITLE
Refactor "app/authorization.go" to use structured logging

### LINKED ISSUE #12195: Refactor "app/authorization.go" to use structured logging
Mattermost is refactoring the logs to be fully structured and we're looking for contributors to help with that effort. This Help Wanted issue is to change the usages of `mlog.` methods with `fmt.Sprintf` to proper `mlog.Field()` calls in file `app/authorization.go`.

The expected way to implement it is to open the file `app/authorization.go`, look for all calls to `mlog.Info`, `mlog.Debug`, etc and in cases where `fmt.Sprintf` is used to write the log - modify the call to use `mlog.Bool`, `mlog.String`, etc to pass the parameters, instead of passing them as part of the format string to `Sprintf`.

Example: https://github.com/mattermost/mattermost-server/pull/12038

----
If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

JIRA: https://mattermost.atlassian.net/browse/MM-18285

### PR DESCRIPTION
**Pull request by sowmiyamuthuraman**
Fixes #12195 

*Ticket link*
https://mattermost.atlassian.net/browse/MM-18285

### CODE REVIEW

**Comment by hanzei** [comment]
@sowmiyamuthuraman You need to format `app/authorization.go` with `go fmt`. Otherwise the CI will fail.

File: app/authorization.go [hunk 1]
```diff
@@ -192,8 +191,7 @@ func (a *App) RolesGrantPermission(roleNames []string, permissionId string) bool
 	if err != nil {
 		// This should only happen if something is very broken. We can't realistically
 		// recover the situation, so deny permission and log an error.
-		mlog.Error("Failed to get roles from database with role names: " + strings.Join(roleNames, ","))
-		mlog.Error(fmt.Sprint(err))
+		mlog.Error("Failed to get roles from database with role names: " + strings.Join(roleNames, ","),mlog.Err(err))
```
Comment (lieut-data) [inline app/authorization.go]: nit: whitespace seems to be off here, missing a space before `mlog.Err`.

File: app/post_test.go [hunk 2]
```diff
@@ -258,22 +258,17 @@ func TestUpdatePostEditAt(t *testing.T) {
...
 
 	post.IsPinned = true
-	if saved, err := th.App.UpdatePost(post, true); err != nil {
-		t.Fatal(err)
-	} else if saved.EditAt != post.EditAt {
-		t.Fatal("shouldn't have updated post.EditAt when pinning post")
-
-		*post = *saved
-	}
+	saved, err := th.App.UpdatePost(post, true)
```
Comment (lieut-data) [inline app/post_test.go]: These improvements look good -- but they don't seem to be related to the PR or ticket. Can you submit these separately?

File: app/authorization.go [hunk 3]
```diff
@@ -192,8 +191,7 @@ func (a *App) RolesGrantPermission(roleNames []string, permissionId string) bool
 	if err != nil {
 		// This should only happen if something is very broken. We can't realistically
 		// recover the situation, so deny permission and log an error.
-		mlog.Error("Failed to get roles from database with role names: " + strings.Join(roleNames, ","))
-		mlog.Error(fmt.Sprint(err))
+		mlog.Error("Failed to get roles from database with role names: "+strings.Join(roleNames, ",")+" ", mlog.Err(err))
```
[the remark this answers is not in the document]
Comment (sowmiyamuthuraman) [reply app/authorization.go]: > Nit: there still seems to be missing space around the `strings.Join` concatenation and I believe the extra space in the string is not needed with the structured logging:

@marianunez gofmt removes the space around the `strings.Join`

[the remark this answers is not in the document]
Comment (marianunez) [reply app/authorization.go]: Weird, it does add the space when there is no extra parameter in the `mlog` 🤔
